Sources/Filters/Core/Cutter/index.js

```diff
@@ -0,0 +1,214 @@
+    // and process each cell.
+    for (let cellId = 0; cellId < numCells; cellId++) {
+      const nbPointsInCell = dataCell[0];
+      // Check that cells have at least 3 points
+      if (nbPointsInCell > 2) {
+        // Get ids of points that constitute the current cell and its assotiated scalar
+        const cellPointsID = [];
+        const cellPointsScalars = [];
+        for (let pointIndex = ((nbPointsInCell + 1) * cellId) + 1;
+          pointIndex < (nbPointsInCell + 1) * (cellId + 1); pointIndex++) {
+          cellPointsID.push(dataCell[pointIndex]);
+          const val = cutScalars.getComponent(dataCell[pointIndex], 0);
+          cellPointsScalars.push(val);
+        }
+
+        // Check if all cell points are on same side (same side == cell not crossed by cut function)
+        const sideFirstPoint = cellPointsScalars[0] > 0;
+        let allPointsSameSide = true;
+        for (let i = 1; i < cellPointsScalars.length; i++) {
+          const sideCurrentPoint = cellPointsScalars[i] > 0;
+          if (sideCurrentPoint !== sideFirstPoint) {
+            allPointsSameSide = false;
+            break;
+          }
+        }
+
+        // Go to next cell if cell is not crossed by cut function
+        if (!allPointsSameSide) {
+          // Find and compute edges which intersect cells
+          const intersectedEdgesList = [];
+          for (let i = 0; i < cellPointsID.length; i++) {
+            const idNext = i + 1 === cellPointsID.length ? 0 : i + 1;
+
+            // Check if edge is crossed
+            const signPoint0 = cellPointsScalars[i] > 0;
+            const signPoint1 = cellPointsScalars[idNext] > 0;
+            if (signPoint1 !== signPoint0) {
+              const vert = [i, idNext];
+              // Compute preferred interpolation direction
+              let deltaScalar = cellPointsScalars[vert[1]] - cellPointsScalars[vert[0]];
+              let e1 = vert[0];
+              let e2 = vert[1];
+              if (deltaScalar <= 0) {
+                e1 = vert[1];
+                e2 = vert[0];
+                deltaScalar *= -1;
+              }
+
+              // linear interpolation
+              let t = 0.0;
+              if (deltaScalar !== 0.0) {
+                t = (model.cutValue - cellPointsScalars[e1]) / deltaScalar;
+              }
+
+              const pointID1 = cellPointsID[e1];
+              const pointID2 = cellPointsID[e2];
+              const x1 = [];
+              const x2 = [];
+              points.getPoint(pointID1, x1);
+              points.getPoint(pointID2, x2);
+
+              // Compute the intersected point on edge
+              const computedIntersectedPoint = [];
+              for (let j = 0; j < 3; j++) {
+                computedIntersectedPoint[j] = x1[j] + (t * (x2[j] - x1[j]));
+              }
+
+              intersectedEdgesList.push({ pointEdge1: pointID1, // id of one point of the edge
+                point2Edge: pointID2, // id of one point of the edge
```

Comment:
@laurennlam that was the source of the issue i showed you (should be pointEdge2 not point2Edge)

Reply:
Thanks for the report @agirault and sorry for this basic error...

Reply:
No problem at all! And no, I have a pr coming up with this and various performance improvements. Thank you!